src/checkers/jsonchecker.py

```diff
@@ -68,10 +71,18 @@ class JSONChecker(HTMLChecker):
     }
     SUPPORTED_DATA_CLASSES = [ExternalData, ExternalGitRepo]
 
-    async def _get_json(self, url: str) -> bytes:
+    async def _get_json(self, url: t.Union[str, URL]) -> bytes:
         log.debug("Get JSON from %s", url)
+        url = URL(url)
+
+        headers = {}
+        if url.host == "api.github.com":
```

Comment:
Here we rely on that yarl reliably parses URLs and can't be tricked into thinking that the host is different from what it really is. Otherwise, we can send our token to a malicious host, since the URL is provided by user.

@barthalion If you consider that this check is not enough (or you're paranoid like me), we can use different tokens for querying GitHub APIs here and opening/merging PRs in main.py.

Reply:
It's fine by me the way it is now, thanks.